### PR TITLE
[SPARK-17412][DOC] All test should not be run by `root` or any admin user

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

`FsHistoryProviderSuite` fails if `root` user runs it. The test case **SPARK-3697: ignore directories that cannot be read** depends on `setReadable(false, false)` to make test data files and expects the number of accessible files is 1. But, `root` can access all files, so it returns 2.

This PR adds the assumption explicitly on doc. `building-spark.md`.
## How was this patch tested?

This is a documentation change.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala [hunk 1]
```diff
@@ -141,7 +141,7 @@ class FsHistoryProviderSuite extends SparkFunSuite with BeforeAndAfter with Matc
 
     val provider = new FsHistoryProvider(createTestConf())
     updateAndCheck(provider) { list =>
-      list.size should be (1)
+      list.size should be (if (System.getProperty("user.name").equals("root")) 2 else 1)
```
Comment (viirya) [inline core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala]: Do we need to consider Windows user?

Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala]: Maybe not yet. I made this since it's reported.

Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala]: Thank you for review, @viirya .

**Comment by dongjoon-hyun** [comment]
Hi, @vanzin , could you review this PR?
The test is written by you here.

https://github.com/apache/spark/commit/3052f4916e7f2c7fbc4837f00f4463b7d0b34718#diff-6d8e0a7cc5dc5b65b81882c3a8e7c370R191

**Comment by vanzin** [comment]
It's probably worth pointing out that no one should really be running any of this as "root"...

**Comment by dongjoon-hyun** [comment]
Yep. It's not recommended, too. In fact, during RC4 pre-preparation with new machine, it happens to me. Because, I installed Java/Python/R and tried this before making any user account.

If you prefer, I can transform this PR into some warning or documentation PR. How do you think about that? I think we had better resolve the current status in any way. Already, a user create an JIRA issue for this.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @vanzin !

**Comment by dongjoon-hyun** [comment]
May I turn this into to raise exception with more clear message like: **`root` is not allowed to test this suite**?

**Comment by vanzin** [comment]
That's what "assume" does, except it doesn't fail the test.

But the larger point is that if you want to change this, you shouldn't just whitelist root, because as has been pointed out, the same problem might exist in other situations.

So either do nothing and just don't run things as root, or disable the test properly so that it covers any non-expected case.

**Comment by dongjoon-hyun** [comment]
Then, to make that assumption clearly, what about adding documents here simply?

http://spark.apache.org/docs/latest/building-spark.html#running-tests

**Comment by vanzin** [comment]
I thought it was pretty obvious that you shouldn't be running things as root or any admin user, but apparently it's not... fine if you want to add it, I just find it unnecessary.

**Comment by dongjoon-hyun** [comment]
Sorry for asking again, @vanzin . It's a little bit confusing to me. To make it sure, are you agreeing on adding a few caution on documentation or saying this kind of PR is unnecessary in any ways (code/docs)?
> fine if you want to add it, I just find it unnecessary.

**Comment by vanzin** [comment]
I'm saying that if you really want to add a warning to the docs, go ahead. I just think that's unnecessary and that it should be obvious to anyone.

Similarly, I'm saying that if you actually want to change the test, make the correct change that will work everywhere, not the change that detects a single case where the test doesn't work.

**Comment by dongjoon-hyun** [comment]
Thank you so much. I think a warning on docs is the best because
- The assumption is valid.
- There might be more test cases assuming on the user rights.
- We can answer in JIRA based on the docs. (We don't need to explain newcomers about the assumption.)

File: docs/building-spark.md [hunk 2]
```diff
@@ -214,7 +214,7 @@ For help in setting up IntelliJ IDEA or Eclipse for Spark development, and troub
 
 # Running Tests
 
-Tests are run by default via the [ScalaTest Maven plugin](http://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin).
+Tests are run by default via the [ScalaTest Maven plugin](http://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin). All tests should not be run by `root` or any admin user.
```
Comment (srowen) [inline docs/building-spark.md]: I'm OK with it though might suggest slightly different wording, like "note that tests should not be run as `root` or an admin user."

**Review by srowen — APPROVED** [review]

File: docs/building-spark.md (hunk 2 above)
Comment (dongjoon-hyun) [inline docs/building-spark.md]: Thank you for review, @srowen . I'll update like that.

**Comment by vanzin** [comment]
Merging to master.

**Comment by dongjoon-hyun** [comment]
Thank you for review and merging, @vanzin, @srowen , @viirya !

**Pull request closed without merging**